resources/js/components/UserAuthForm.vue

```diff
@@ -0,0 +1,266 @@
+                class="appearance-none block w-full px-3 py-2 border border-gray-300 rounded-md shadow-sm placeholder-gray-400 focus:outline-none focus:ring-indigo-500 focus:border-indigo-500 sm:text-sm"
+              />
+            </div>
+            <div v-if="error.rePassword" class="mt-2">
+                <span class="error-text">{{error.rePassword}}</span>
+            </div>
+          </div>
+
+          <div v-if="!isSignup" class="flex items-center justify-between">
+            <div class="flex items-center">
+              <input
+                id="remember_me"
+                name="remember_me"
+                type="checkbox"
+                v-model="userInfo.remember_me"
+                class="h-4 w-4 text-indigo-600 focus:ring-indigo-500 border-gray-300 rounded"
+              />
+              <label for="remember_me" class="ml-2 block text-sm text-gray-900">
+                Remember me
+              </label>
+            </div>
+
+            <div class="text-sm">
+              <a
+                href="/password/reset"
+                class="font-medium text-indigo-600 hover:text-indigo-500"
+              >
+                Forgot your password?
+              </a>
+            </div>
+          </div>
+
+          <div>
+            <button
+              type="submit"
+              class="w-full flex justify-center py-2 px-4 border border-transparent rounded-md shadow-sm text-sm font-medium text-white bg-indigo-600 hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
+            >
+              {{ buttonText }}
+            </button>
+          </div>
+        </form>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import axios from 'axios';
+import VueToast from 'vue-toast-notification';
+
+export default {
+  data() {
+    return {
+      userInfo: {
+        name: "",
+        email: "",
+        password: "",
+        image: null,
+        remember_me: false
+      },
+
+      rePassword: "",
+
+      error: {},
+    };
+  },
+
+  props: ["buttonText", "isSignup"],
+
+  methods: {
+    validateEmail: function (email) {
+      const re = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+      return re.test(String(email).toLowerCase());
+    },
+
+    getBase64(file) {
+      return new Promise((resolve, reject) => {
+        const reader = new FileReader();
+        reader.readAsDataURL(file);
+        reader.onload = () => resolve(reader.result);
+        reader.onerror = error => reject(error);
+      });
+    },
+
+    setFile([file]){
+      this.error.image = "";
+      this.getBase64(file).then(data => this.userInfo.image = data);
+    },
+
+    async signUp(){
+      try{
+        const response = await axios.post('/api/auth/signup',this.userInfo);
+        Vue.$toast.success(response.data.message);
+        window.location.href = '/';
```

Comment:
Evaluar utilizar el router de Vue.js. Quedaria algo asi:
this.$router.push({ name: 'Index' });